planet/Objects.ocd/Libraries.ocd/Structures.ocd/PlayerKnowledge.ocd/Script.c

```diff
@@ -2,76 +2,89 @@
 	Player Knowledge
 
 	Library to control the players knowledge/plans:
-     * GetPlrKnowledge(int player, id plan, int index, int category)
-     * HasPlrKnowledge(int player, id plan)
-     * SetPlrKnowledge(int player, id plan, bool remove)
-     * GivePlrKnowledge(int player, id plan)
-     * RemovePlrKnowledge(int player, id plan)
+     * GetPlayerKnowledge(int player, int index, int category)
+     * HasPlayerKnowledge(int player, id plan)
+     * SetPlayerKnowledge(int player, id plan, bool remove)
+     * GivePlayerKnowledge(int player, id plan)
+     * RemovePlayerKnowledge(int player, id plan)
 
     Should be incorporated into a player management system later
 
 	@author Marky
 */
 
-/* --- Properties --- */
-
-// Local variables to store the player's knowledge.
-local knowledge = [];
 
+func Definition(id type)
+{
+	// As soon as the definition is loaded, we define a custom data topic in the player data
+	PlayerData->GetData(type).knowledge = [];
```

Comment:
I don't understand this entirely - is this a single global array for all players or is it copied for individual players somewhere?

Reply:
Argh, you might be right... the array is added to the prototype, but if a new instance of the proplist is instantiated it will still reference the global array in the prototype, instead of creating its own empty array, right?